cost/cloud_cost_anomaly_alerts/cloud_cost_anomaly_alerts.pt

```diff
@@ -196,39 +196,52 @@ script "js_filter_dimensions", type: "javascript" do
-      exist = false;
-      filter = {
-        "dimensionid":"",
-        "dimensionname":"",
-        "type":"",
-        "value":""
-      }
-      dimension_filter_map = user_input_dimension.split("=")
-      hasfilter = false;
-      if (dimension_filter_map.length > 1){
-        hasfilter=true
-      }
-      _.each(ds_dimensions, function(dimension) {
-        if (dimension_filter_map[0] === dimension.name || dimension_filter_map[0] === dimension.id) {
-          filtered_dimensions.ids.push(dimension.id);
-          filtered_dimensions.names.push(dimension.name);
-          exist = true;
-          if (hasfilter) {
-            filter.dimensionid=dimension.id;
-            filter.dimensionname=dimension.name;
-            filter.type="equal";
-            filter.value=dimension_filter_map[1];
-            filtered_dimensions.filters.push(filter);
+    ids = _.map(ds_dimensions, function(value){ return value.id});
+    names = _.map(ds_dimensions, function(value){ return value.name});
+
+    _.each(param_dimensions, function (user_input_dimension) {
+      input_map = user_input_dimension.split("=")
+      if (!(exists = _.contains(ids, input_map[0]) != false ? true : _.contains(names, input_map[0]) != false ? true : false)){
+        filtered_dimensions.invalid_ids.push(input_map[0])
+      }else{
+        var dimensionid = ""
+        var dimensionname = ""
+        if (ids.indexOf(input_map[0]) > -1){
+          element = ids.indexOf(input_map[0])
+          dimensionname = names[element]
+          dimensionid = ids[element]
+          filtered_dimensions.ids.push(dimensionid)
+          filtered_dimensions.names.push(dimensionname)
+        }else {
+          element = names.indexOf(input_map[0])
+          dimensionname = names[element]
+          dimensionid = ids[element]
+          filtered_dimensions.ids.push(dimensionid)
+          filtered_dimensions.names.push(dimensionname)
+        }
+        if (input_map.length > 1) {
+          multifiltervalues = input_map[1].split(",");
+          switch (true) {
+            case multifiltervalues.length === 1 :
+              filter = { "dimensionid":dimensionid, "dimensionname":dimensionname, "value": multifiltervalues[0], "expressiontype": "and"}
+              expression = {"type": "equal", "dimension": dimensionid, "value": multifiltervalues[0]}
+              filtered_dimensions.filters.push(filter)
+              filtered_dimensions.expressionsand.push(expression)
+              break;
+            case multifiltervalues.length > 1 :
+              _.each(multifiltervalues, function (value) {
+                filter = { "dimensionid":dimensionid, "dimensionname":dimensionname, "value": value, "expressiontype": "or"}
+                expression = {"type": "equal", "dimension": dimensionid, "value": value}
+                filtered_dimensions.filters.push(filter)
+                filtered_dimensions.expressionsor.push(expression)
+              });
+              break;
```

Comment:
Both cases are almost the same, the only thing that changes is `"expressiontype"` in `filter` object and where you push the `filter` object, probably we can do something like this:
```
var expressiontype = ""

if (multifiltervalues.length === 1) {
   expressiontype = "and"
} else if (multifiltervalues.length > 1) {
   expressiontype = "or"
}

_.each(multifiltervalues, function (value) {
  filter = {"dimensionid":dimensionid, "dimensionname":dimensionname, "value": value, "expressiontype": expressiontype}
  expression = {"type": "equal", "dimension": dimensionid, "value": value}
  filtered_dimensions.filters.push(filter)
  
  if (expressiontype === "and") {
     filtered_dimensions.expressionsand.push(expression)
  } else if (expressiontype === "or") {
     filtered_dimensions.expressionsor.push(expression)
  }
});
```